src/Interpreters/TreeRewriter.cpp

```diff
@@ -325,6 +326,39 @@ struct ExistsExpressionData
 
 using ExistsExpressionVisitor = InDepthNodeVisitor<OneTypeMatcher<ExistsExpressionData>, false>;
 
+struct ReplacePositionalArgumentsData
+{
+    using TypeToVisit = ASTSelectQuery;
+    ContextPtr context;
+
+    void visit(ASTSelectQuery & select_query, ASTPtr &) const
+    {
+        if (context->getSettingsRef().enable_positional_arguments)
+        {
+            if (select_query.groupBy())
+            {
+                for (auto & expr : select_query.groupBy()->children)
+                    replaceForPositionalArguments(expr, &select_query, ASTSelectQuery::Expression::GROUP_BY);
+            }
+            if (select_query.orderBy())
+            {
+                for (auto & expr : select_query.orderBy()->children)
+                {
+                    for (auto & elem : assert_cast<ASTOrderByElement &>(*expr).children)
```

Comment:
The fix is actually only in this line (extracting child of ASTOrderByElement). It was not necessary to add a visitor - it works in subquery without it. But it is now more consistent with the other code.